diff --git a/drivers/input/touchscreen/omnivision_tcm/omnivision_tcm_i2c.c b/drivers/input/touchscreen/omnivision_tcm/omnivision_tcm_i2c.c
--- a/drivers/input/touchscreen/omnivision_tcm/omnivision_tcm_i2c.c
+++ b/drivers/input/touchscreen/omnivision_tcm/omnivision_tcm_i2c.c
@@ -231,6 +231,6 @@ static int ovt_tcm_i2c_rmi_read(struct ovt_tcm_hcd *tcm_hcd,
 		unsigned short addr, unsigned char *data, unsigned int length)
 {
-	int retval;
+	int retval = -EIO;
 	unsigned char address;
 	unsigned int attempt;
 	struct i2c_msg msg[2];
@@ -278,6 +278,6 @@ static int ovt_tcm_i2c_rmi_write(struct ovt_tcm_hcd *tcm_hcd,
 		unsigned short addr, unsigned char *data, unsigned int length)
 {
-	int retval;
+	int retval = -EIO;
 	unsigned int attempt;
 	unsigned int byte_count;
 	struct i2c_msg msg;
@@ -335,6 +335,6 @@ static int ovt_tcm_i2c_read(struct ovt_tcm_hcd *tcm_hcd, unsigned char *data,
 		unsigned int length)
 {
-	int retval;
+	int retval = -EIO;
 	unsigned int attempt;
 	struct i2c_msg msg;
 	struct i2c_client *i2c = to_i2c_client(tcm_hcd->pdev->dev.parent);
@@ -373,6 +373,6 @@ static int ovt_tcm_i2c_write(struct ovt_tcm_hcd *tcm_hcd, unsigned char *data,
 		unsigned int length)
 {
-	int retval;
+	int retval = -EIO;
 	unsigned int attempt;
 	struct i2c_msg msg;
 	struct i2c_client *i2c = to_i2c_client(tcm_hcd->pdev->dev.parent);
